flax/linen/fp8_ops.py

```diff
@@ -17,24 +17,36 @@
 
 from flax.linen import initializers
 from flax.linen.module import Module
-from jax import custom_vjp
+from jax import custom_vjp, custom_jvp
 from jax import lax
 from jax import numpy as jnp
 from jax import random
 
 
 # Type annotations
 Array = jnp.ndarray
-Dtype = jnp.dtype
+DType = jnp.dtype
 PRNGKey = jnp.ndarray
 
 class FP8Helper:
+  """
+  FP8 helper to manage the FP8 meta
+  """
+  FWD_DTYPE: DType = jnp.float8_e4m3fn
```

Comment:
I didn't observe the usage of FWD/BWD_DTYPE in this commit. Perhaps we should limit this commit to handling accumulation datatypes exclusively. For simplicity, I suggest hardcoding the accumulation datatypes directly within the dot general custom gradient functions as we have already done for the input/output dtypes.